core/app/models/spree/address.rb

```diff
@@ -225,5 +226,12 @@ def postal_code_validate
 
       errors.add(:zipcode, :invalid) unless ::ValidatesZipcode.valid?(formatted_zip, country_iso.upcase)
     end
+
+    def assign_new_default_address_to_user
+      return unless user
+
+      user.update(bill_address: user.addresses.last) if user.bill_address == self
+      user.update(ship_address: user.addresses.last) if user.ship_address == self
```

Comment:
```suggestion
      user.update(bill_address: user.addresses.last) if user.bill_address == self
      user.update(ship_address: user.addresses.last) if user.ship_address == self
```

@KacperMekarski let's not do 2 seperate SQL queries, we can assign attributes to object and do one `.save!`

Reply:
done!